### ISSUE TITLE
License confusion

### ISSUE THREAD

**Issue by RoGryza** [NorfairKing/intray#9]
I noticed Intray isn't published under a free license, am I allowed to fork it/publish changes? Also, would a PR for Postgres support be welcome?

**Comment by NorfairKing** [comment]
Hi! I thought I published Intray under a free license... Hmm
I will have to choose one then.
I'd like to try out this arrangement of "free if you're not making money off it, but you can pay for a license where you can".
I'm just not sure how that works.

For a Postgres PR: I'd like to know where this idea came from.
I chose sqlite because it's the simplest thing I could get working and the database is _tiny_ in practice because every row started as a user action.
If we were to support postgres, I would have to support it, so you'd have to get all the integration tests to work with postgres as well.
This is certainly not trivial, setup-wise.

TL;DR: Why would you want postgres support?

**Comment by RoGryza** [comment]
I'm not very knowledgeable on licensing but I'd look into dual-licensed software like Qt or Sencha Ext JS.

I wanted to use Postgres I'll have a replicated instance setup for other stuff so using it for Intray would mean I don't need to worry about backing up the SQLite database.

**Comment by NorfairKing** [comment]
Thanks for the tip, I'll have a look!

> I wanted to use Postgres because I'll have a replicated instance setup for other stuff so using it for Intray would mean I don't need to worry about backing up the SQLite database.

That makes a lot of sense, but I'm pretty sure that setting up backups for the sqlite database will be less work than supporting postgres.

If you still want to get postgres support then I'd be happy to review a PR, but again: all the integration tests also have to run against a postgres instance automatically in order for me to accept the PR because I don't run any postgres instances anywhere.

**Comment by RoGryza** [comment]
I'll probably do it then. Setting up Postgres for integration tests is a pain but I've done it at least twice at work recently so hopefully I've figured it out by now

**Comment by NorfairKing** [comment]
@RoGryza Just before you begin, so I'm being extra clear.
`stack test` needs to work without any other setup, and with a clean database for every property-test.
Let's be clear about the approach so that you don't end up doing wasted effort.

**Comment by RoGryza** [comment]
I was thinking of using something like https://hackage.haskell.org/package/tmp-postgres-1.32.0.1/docs/Database-Postgres-Temp.html. I'll still have a look at the source code and, if needed, roll my own.

The postgres and initdb binaries will be required though, is that acceptable?

**Comment by NorfairKing** [comment]
> I was thinking of using something like https://hackage.haskell.org/package/tmp-postgres-1.32.0.1/docs/Database-Postgres-Temp.html. I'll still have a look at the source code and, if needed, roll my own.

That looks good!

> The postgres and initdb binaries will be required though, is that acceptable?

Nono that's fine. As long as we don't have to run any commands beforehand, it'll work.

**Comment by RoGryza** [comment]
@NorfairKing I'm closing this issue since I changed my mind and I'll just use SQLite

**Comment by NorfairKing** [comment]
@RoGryza That's fine by me, but would you please elaborate on why?

**Comment by RoGryza** [comment]
I'm not going to set up a replicated Postgres cluster anymore, so SQLite would work just as well as Postgres for my needs.
